lib/core/presentation/widgets/common/circular_loading/circular_loading_widget.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:math';
+import 'package:app/theme/color.dart';
+import 'package:app/theme/sizing.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+class CircularLoading extends StatefulWidget {
+  const CircularLoading({super.key});
+
+  @override
+  State<CircularLoading> createState() => CircularLoadingState();
+}
+
+class CircularLoadingState extends State<CircularLoading>
+    with TickerProviderStateMixin {
+  late AnimationController _animationContrl;
+  late Animation<double> _animation;
+
+  @override
+  void initState() {
```

Comment:
sr if you could also add dispose method , and animationCtrl.dispose  inside it ?